src/passes/OptimizeInstructions.cpp

```diff
@@ -110,6 +115,7 @@ struct LocalScanner : PostWalker<LocalScanner> {
     auto* value =
       Properties::getFallthrough(curr->value, passOptions, *getModule());
     auto& info = localInfo[curr->index];
+    info.minBits = std::min(info.minBits, Bits::getMinBits(value, this));
     info.maxBits = std::max(info.maxBits, Bits::getMaxBits(value, this));
```

Comment:
@kripken I guess it could be more precise. Something like this:
```c++
auto valueMinBits = Bits::getMinBits(value, this);
if (valueMinBits == 0 || info.minBits == 0) {
  info.minBits = 0;
} else {
  info.minBits = valueMinBits;
}

auto valueMaxBits = Bits::getMaxBits(value, this);
auto typeMaxBits = value->type.getByteSize() * 8;
if (valueMaxBits == typeMaxBits || info.maxBits == typeMaxBits) {
  info.maxBits = typeMaxBits;
} else {
  info.maxBits = valueMaxBits;
}
```
is it correct?

Reply:
Hmm, 0 isn't "unknown", it is that there may be 0 bits set. I don't think we can add those ifs. Say `foo() ? 0 : bar()` will have minBits as 0, as 0 is a possible value there. We can't ignore that later.

Reply:
So perhaps we can reuse `kUnknown` with similar finalization stage as it's done for `signExtedBits`?

Reply:
Also, may be `select` should be handled separately?